tmt/utils.py

```diff
@@ -1006,8 +1006,93 @@ def environment_to_dict(variables: Union[str, List[str]]) -> EnvironmentType:
     return result
 
 
-def environment_file_to_dict(
-        env_files: Iterable[str], root: str = ".") -> Dict[str, str]:
+@lru_cache(maxsize=None)
+def environment_file_to_dict(env_file: str, root: str = ".") -> EnvironmentType:
```

Comment:
We break API (renamed argument) but no one seems to use that outside of tmt/base.py

Reply:
I replaced all calls to `environment_file_to_dict` with new `environment_files_to_dict`, to match the name, purpose, and signature. Code in `tmt/base.py` should be perfectly happy, since it['s not calling `environment_file_to_dict` anymore.